### PR TITLE
LPS-155286 Cannot POST or PUT Region without "position" field

### PR DESCRIPTION
**Pull request by RafaelCRC**

### CODE REVIEW

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@RafaelCRC the changes look good, but we also need test coverage for this. You can force-push, or if you'd like to close and re-send that is fine too.

**Comment by epark18** [comment]
> ### ✔️ ci:test:stable - 26 out of 26 jobs passed
> ### ❌ ci:test:relevant - 47 out of 54 jobs passed in 1 hour 8 minutes
CI failures are unrelated :]

**Comment by RafaelCRC** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java [hunk 1]
```diff
@@ -356,6 +365,28 @@ private <T extends Exception> void _testPostCountryRegionProblem(
 		}
 	}
 
+	private void _testPostPositionEmpty(Region region) throws Exception {
+		Region regionReturn = _addRegion(region);
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java:369]: I would call `regionResource.postRegion` here. It's better to be explicit about it if possible.

File: modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java [hunk 2]
```diff
@@ -170,6 +175,10 @@ public void testPutRegion() throws Exception {
 
 		Region randomRegion = randomRegion();
 
+		randomRegion.setPosition((Double)null);
+
+		_testPutPositionEmpty(region1.getId(), randomRegion);
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java:180]: Same idea as with post: 

- [ ] move this assertion to the bottom, after the required fields
- [ ] inline the helper method

**Review by drewbrokke — CHANGES_REQUESTED** [review]
Hey @RafaelCRC this is looking good, just a few adjustments needed to make things more consistent.

File: modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java [hunk 3]
```diff
@@ -136,29 +137,42 @@ public void testPostCountryRegion() throws Exception {
...
+		region1.setRegionCode("");
 
-		_testPostCountryRegionProblem(region, null);
+		_testPostCountryRegionProblem(region1, null);
+
+		region1.setRegionCode(existingRegion.getRegionCode());
+
+		_testPostCountryRegionProblem(region1, DuplicateRegionException.class);
+
+		Region region2 = randomRegion();
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java:164]: It's fine to re-use `region1` here. Then we can just keep the original variable name of `region`.

File: modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java [hunk 4]
```diff
@@ -136,29 +137,42 @@ public void testPostCountryRegion() throws Exception {
...
+		Region regionReturn = regionResource.postCountryRegion(
+			_country.getCountryId(), region2);
 
-		region.setRegionCode(existingRegion.getRegionCode());
+		double regionPosition = regionReturn.getPosition();
 
-		_testPostCountryRegionProblem(region, DuplicateRegionException.class);
+		Assert.assertEquals(
+			String.valueOf(regionPosition),
+			String.valueOf(GetterUtil.DEFAULT_DOUBLE));
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java:175]: Can we do without casting to a string here? Just compare the double values directly?

File: modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java [hunk 5]
```diff
@@ -136,29 +137,42 @@ public void testPostCountryRegion() throws Exception {
...
 
-		region.setRegionCode("");
+		region2.setPosition((Double)null);
 
-		_testPostCountryRegionProblem(region, null);
+		Region regionReturn = regionResource.postCountryRegion(
+			_country.getCountryId(), region2);
 
-		region.setRegionCode(existingRegion.getRegionCode());
+		double regionPosition = regionReturn.getPosition();
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java:171]: SF: inline this variable

File: modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java [hunk 6]
```diff
@@ -194,6 +208,19 @@ public void testPutRegion() throws Exception {
 
 		_testPutRegionProblem(
 			region1.getId(), randomRegion, DuplicateRegionException.class);
+
+		Region region3 = randomRegion();
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java:212]: You can re-assign the `randomRegion` variable here too.

File: modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java [hunk 7]
```diff
@@ -194,6 +208,19 @@ public void testPutRegion() throws Exception {
...
+
+		region3.setPosition((Double)null);
+
+		Region regionReturn = regionResource.putRegion(
+			region1.getId(), region3);
+
+		double regionPosition = regionReturn.getPosition();
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-address/headless-admin-address-test/src/testIntegration/java/com/liferay/headless/admin/address/resource/v1_0/test/RegionResourceTest.java:219]: SF: inline this variable.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
So far so good, just a little more tightening up to do.

**Review by drewbrokke — APPROVED** [review]

**Comment by drewbrokke** [comment]
ci:forward

**Pull request closed without merging**